xpresso/applications.py

```diff
@@ -201,6 +208,9 @@ def _setup(self) -> typing.List[typing.Callable[..., typing.AsyncIterator[None]]
                         ],
                         container=self.container,
                     )
+                    for dep in operation.dependant.get_flat_subdependants():
+                        if dep.scope == "app":
+                            lifespan_dependants.append(dep)
```

Comment:
This is an interesting bug!

By default `"app"` scoped dependencies are lazy-loaded.
This means that `anyio.create_task_group` isn't called until it's needed, by the endpoint function.
So it gets called from a _request_ event loop task, _not_ from the lifespan event loop task.
Anyio detects this and reasonably prohibits it.

The fix is pretty straightforward: collect and initialize all `"app"` scoped dependencies from within the lifespan.